scarb/src/ops/resolve.rs

```diff
@@ -183,3 +167,25 @@ async fn collect_packages_from_resolve_graph(
 fn build_cfg_set(target: &Target) -> CfgSet {
     CfgSet::from_iter([Cfg::kv("target", target.kind.clone())])
 }
+
+#[tracing::instrument(skip_all, level = "debug")]
+pub fn is_cairo_version_compatible(
+    packages: &Vec<Package>,
```

Comment:
```suggestion
    packages: &[Package],
```

passing slices is favourable over passing reference to a vector